### PR TITLE
Ignore MSVC warnings (via Gisle Vanem)

### PR DESCRIPTION
**Pull request by richsalz**

### CODE REVIEW

File: include/openssl/lhash.h [hunk 1]
```diff
@@ -192,11 +192,24 @@ void OPENSSL_LH_node_usage_stats_bio(const OPENSSL_LHASH *lh, BIO *out);
     { \
         OPENSSL_LH_doall_arg((OPENSSL_LHASH *)lh, (OPENSSL_LH_DOALL_FUNCARG)fn, (void *)arg); \
     } \
-    LHASH_OF(type)
+LHASH_OF(type)
```
Comment (levitte) [inline include/openssl/lhash.h]: Uhmmm, no.  It was correctly aligned with the previous macro lines.

**Comment by dot-asm** [comment]
Working around compiler warning bugs is source code is -1 as far as I'm concerned. But I don't understand, this warning is suppressed through Configurations/10-main.cf...

**Comment by dot-asm** [comment]
> Working around compiler warning bugs is source code is -1 as far as I'm concerned.

Well, this is not entirely true. Let me put it this way. I'm opposed to working around compiler warning bugs *with explicit reference to specific compiler* in source code, e.g. like here with #ifdef and compiler-specific #pragmas. As general rule compiler bugs needs to be fixed, not worked around. There might be exclusions in cases compiler is wide-spread and users' choice is limited, but this doesn't really apply to warnings.

**Comment by levitte** [comment]
Hasn't this particular issue been rehashed a number of times, ending with a *can't fix* comment, because fixing would result in backward compatibility issues the size of Godzilla?  I do recall trying to fix the bug at some point, ending up with a monumental entanglement...

So it seems that *for now*, this pragma is needed, or putting up with the annoying warning (and subjecting every one else to it as well, with the obvious rehashing as a result).  It's a little like the pointer size issues we get with VMS, which has also been solved with compiler specific pragmas ;-) (although I'll agree, those issue *are* impossible to get around any other way)

... that is, unless you have an actual fix up your sleeve.  In that case, by all means show!

**Comment by levitte** [comment]
> But I don't understand, this warning is suppressed through Configurations/10-main.cf...

Because it's just for *our* building, it doesn't affect everyone else that do `#include <openssl/lhash.h>` or including anything that includes that one.

And ah, in my comment above about pragmas, I forgot for a moment that this is a public header.  I do understand your concern a bit better...  not saying that I agree in this case, though.

**Comment by t-j-h** [comment]
@dot-asm make it clear if this is a -1 so we can resolve via team vote or just a comment on your preference. In this context it is the right "solution" as leaving warnings enabled and fatal is how code should be built - and leaving this as is encourages ignoring warnings completely.

**Comment by richsalz** [comment]
To emphasize Richard's point:  this is a public header file, and it's not just *our* code that gets the warnings, but anything else built on windows that uses this header, such as curl.

File: include/openssl/lhash.h (hunk 1 above)
Comment (levitte) [inline include/openssl/lhash.h]: Ping

**Comment by richsalz** [comment]
$DAYJOB got in the way :)  Updated commit pushed that fixes the indent.

**Comment by richsalz** [comment]
975b993 in 1.1.0 and 05004f3 in master; thanks!

**Comment by dot-asm** [comment]
>leaving this as is encourages ignoring warnings completely.

As well as letting vendor not fixing bug. It's double-edged.

**Comment by levitte** [comment]
> As well as letting vendor not fixing bug. It's double-edged.

I'll admit I've still not quite understood how this is a compiler bug...  'cause that's what you're saying, right?

**Comment by FdaSilvaYY** [comment]
#1939 and #2214 are related to this PR ; it has to wait for 1.2.0 for a whole fix

**Pull request closed without merging**